File: problem_solutions/fibonacci-number/recursive_solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int fib(int n) {
        // base case
        if (n == 0) return 0;
        if (n == 1) return 1;

        // transition
        return fib(n - 1) + fib(n - 2);
    }
};

int main() {
    Solution sol;
    int n = 7;
    cout << sol.fib(n) << endl;

    return 0;
}

File: problem_solutions/fibonacci-number/recursive_solution.py
class Solution:
    def fib(self, n: int) -> int:
        # base case
        if n == 0:
            return 0
        if n == 1:
            return 1

        # transition
        return self.fib(n - 1) + self.fib(n - 2)


if __name__ == "__main__":
    sol = Solution()
    n = 7
    print(sol.fib(n))
